apiserver/charms.go

```diff
@@ -15,92 +15,36 @@ import (
 	"github.com/juju/errors"
 
 	"github.com/juju/juju/internal/charm"
-	"github.com/juju/juju/internal/charm/downloader"
-	"github.com/juju/juju/internal/charm/services"
-	"github.com/juju/juju/state"
 )
 
-// CharmUploader is an interface that is used to update the charm in
-// state and upload it to the object store.
-type CharmUploader interface {
-	UpdateUploadedCharm(info state.CharmInfo) (services.UploadedCharm, error)
-	PrepareCharmUpload(curl string) (services.UploadedCharm, error)
-	ModelUUID() string
-}
-
-// RepackageAndUploadCharm expands the given charm archive to a
-// temporary directory, repackages it with the given curl's revision,
-// then uploads it to storage, and finally updates the state.
-func RepackageAndUploadCharm(
-	ctx context.Context,
-	objectStore services.Storage,
-	uploader CharmUploader,
-	archive *charm.CharmArchive,
-	curl string,
-	charmRevision int,
-) (charm.Charm, string, string, string, error) {
+func repackageCharmWithRevision(ctx context.Context, archive *charm.CharmArchive, charmRevision int) (*charm.CharmArchive, bytes.Buffer, string, error) {
 	// Create a temp dir to contain the extracted charm dir.
 	tempDir, err := os.MkdirTemp("", "charm-download")
 	if err != nil {
-		return nil, "", "", "", errors.Annotate(err, "cannot create temp directory")
+		return archive, bytes.Buffer{}, "", errors.Annotate(err, "cannot create temp directory")
```

Comment:
**suggestion:** We are moving out juju errors. Maybe it's an opportunity to use `juju.Errorf` from internal errors.

(same for all Annotate in this file)

Reply:
Done